fgpyo/sam/__init__.py

```diff
@@ -1231,3 +1263,212 @@ class SamOrder(enum.Enum):
     Coordinate = "coordinate"  #: coordinate sorted
     QueryName = "queryname"  #: queryname sorted
     Unknown = "unknown"  # Unknown SAM / BAM / CRAM sort order
+
+
+@dataclass(frozen=True)
+class SecondaryAlignment:
+    """A secondary alignment as encoded in one part of the `XA` or `XB` tag value on a SAM record.
+
+    Format of a single secondary alignment in the `XA` tag (`,`-delimited):
+
+    ```text
+    chr,<orientation><position>,cigar,NM
+    ```
```

Comment:
**question** `fgpyo` already provides a `SupplementaryAlignment` class for parsing the `XA` tag. I've run into a few cases where it'd be helpful to have an analogous `SecondaryAlignment` class, and I think this PR will be a welcome addition. What are the scenarios where it would make sense to use `SecondaryAlignment` instead of `SupplementaryAlignment` to parse the `XA` tag? 

I do find it sensible to have separate classes to represent the distinct tag formats and the distinct semantic meanings between the two types of alignment, but I'm wondering if there are use cases where it'd be helpful to process both types of alignment in tandem.

https://github.com/fulcrumgenomics/fgpyo/blob/bd3889807834f70078e35f78994acce54eefbe2b/fgpyo/sam/__init__.py#L768-L769

Reply:
All good points! FYI the `SupplementaryAlignment` class is meant to parse the `SA` tag and the `SecondaryAlignment` classes parses the two "sibling" tags that bwa makes for secondaries (`XA` and `XB`).